### PR TITLE
form page 2

### PR DESCRIPTION
**Pull request by ymloh**
[Trello card](https://trello.com/c/UHAd6AMw/165-dev-sml-update-page-2-of-form-to-reflect-updated-wireframe)

WIP: May need tweaking or possibly even a complete do-over.

Couple of new components introduced:
1. [BackContinueButtons](https://github.com/ymloh/council-emissions-calculator-spike/blob/2450eddc28842c347e1f76461037ed0cde398077/pages/form/components/BackContinueButtons.js) - hopefully reusable on all pages of the form.
2. [PredictiveInput](https://github.com/ymloh/council-emissions-calculator-spike/blob/2450eddc28842c347e1f76461037ed0cde398077/pages/form/components/PredictiveInput.js) (see attached video) - takes an array of valid input options, returns an input field that pops up an emulated dropdown menu of valid options, highlighting the first closest match (or none if there's no match), and replacing the entered value with the highlighted match upon losing focus or <Enter> being pressed (or an empty string if there is no match)

modesOfTransport (in constants) has been updated to reflect the new options for mode of transport.

https://user-images.githubusercontent.com/20964807/132002315-24c092ac-dda5-4df9-9b8d-d833fe134c7f.mp4

### CODE REVIEW

**Comment by hqtan** [comment]
Hi @ymloh,

Thx for introducing the new components! :bow: 

can I ask you to merge in the lastest changes from [`CodeforAustralia:main`](https://github.com/CodeforAustralia/council-emissions-calculator-spike) branch to resolve the merge conflicts I'm seeing? e.g. `pages/form/2.js` has been replaced with `page/form/Question2.js`...

**Comment by hqtan** [comment]
nice work on the PR so far @ymloh !
Don't forget to make sure that your PR is up-to-date with the latest changes in [`CodeforAustralia:main`](https://github.com/CodeforAustralia/council-emissions-calculator-spike) branch k? For one, I can see that the form pages haven't been renamed (`pages/form/2.js` should now be `pages/form/Question2.js`), and the progress bar changes seem to not be there...

**Comment by ymloh** [comment]
Thanks for reminding me @hqtan!

I just realised it's not prefilling with the saved answer when the page loads, so I'll work on that next.

**Comment by hqtan** [comment]
@ymloh ,

Are you essentially implementing a combobox with the [`PredictiveInput` component](https://github.com/ymloh/council-emissions-calculator-spike/blob/2450eddc28842c347e1f76461037ed0cde398077/pages/form/components/PredictiveInput.js) ?

If that is the case, I wonder if it's better to use an existing library to provide that functionality, rather than 're-inventing the wheel' ?

Google recommended a couple of libraries that may fit our use-case:
- [Downshift's `useCombobox` React hook](https://downshift.netlify.app/use-combobox#basic-usage)
- [Adobe React Aria's `useCombobox` component](https://react-spectrum.adobe.com/react-aria/useComboBox.html)

I also found this [article about using Chakra-ui with Downshift](https://blog.logrocket.com/building-accessible-components-with-downshift/) that may be useful for our use-case too.

What do you think @ymloh, @batteries76 ?

File: pages/form/Question2.js [hunk 1]
```diff
@@ -1,76 +1,46 @@
...
+        <Heading as="h1" size="md">How do you travel to work in an average week?</Heading>
+        <Text my={5}>Select the main way you travel to work.</Text>
+        <Text my={5}>For example, if you usually drive 2km to the train and then catch the train for 15km, choose train as your way of travel.</Text>
+        <Text my={5}>If you currently work from home, we will use this information to calculate the emissions you save by working at home.</Text>
+        <PredictiveInput options={modesOfTransport} callback={saveAnswer}/>
+      </Box>
+      <BackContinueButtons
+        prevOptions={{
+          href: "/form/Question1",
+          // onClick: saveAnswers,
```
Comment (batteries76) [inline pages/form/Question2.js:35]: Uncomment?

File: pages/form/Question2.js [hunk 2]
```diff
@@ -1,76 +1,46 @@
...
+          disabled: false
+        }}
+        nextOptions={{
+          href: "/form/Question3",
+          // onClick: saveAnswers,
```
Comment (batteries76) [inline pages/form/Question2.js:40]: Uncomment?

File: pages/form/components/BackContinueButtons.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  const PrevButton = () => {
+    if (prevOptions) {
+      return (
+        <LinkButton
+          href={prevOptions.href}
+          onClick={prevOptions.onClick}
+          disabled={prevOptions.disabled}
+          variant="outline"
+        >
+          <svg pwidth="23" height="14" viewBox="0 0 23 14" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (batteries76) [inline pages/form/components/BackContinueButtons.js:41]: Maybe use SVG package?

**Review by batteries76 — COMMENTED** [review]
Good as this is, I think this can be closed for clarity for now, and we can revisit as things progress.
Thanks Meng.

**Pull request closed without merging**